plugins/auto-sizes/hooks.php

```diff
@@ -78,6 +78,26 @@ function auto_sizes_update_content_img_tag( $html ): string {
 }
 add_filter( 'wp_content_img_tag', 'auto_sizes_update_content_img_tag' );
 
+/**
+ * Checks whether the given 'sizes' attribute includes the 'auto' keyword.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $sizes_attr The 'sizes' attribute value.
+ * @return bool True if the 'auto' keyword is present, false otherwise.
+ */
+function auto_sizes_attribute_includes_auto( string $sizes_attr ): bool {
+	if ( str_contains( $sizes_attr, 'auto,' ) ) {
+		return true;
+	}
+
+	if ( 'auto' === $sizes_attr ) {
+		return true;
+	}
+
+	return str_ends_with( $sizes_attr, ' auto' ) || str_ends_with( $sizes_attr, ',auto' );
```

Comment:
What about splitting the string on commas+whitespace and then seeing if `auto` is inside? 

```suggestion
	return in_array( 
		'auto', 
		preg_split( '/\s*,\s*/', strtolower( trim( $sizes_attr ) ) ), 
		true 
	);
```

Note also the inclusion of `strtolower()` since the specs ays it is case-insensitive: https://html.spec.whatwg.org/multipage/images.html#sizes-attributes

Reply:
Note that `\s` may be locale-sensitive, while `" \t\f\r\n"` will always match the rules from HTML and CSS.

if performance is a concern, we could avoid the allocations that `preg_split()` introduces and check directly with a `preg_match`. according to [the spec](https://html.spec.whatwg.org/#parse-a-sizes-attribute), I believe this should indicate if `auto` is already the indicated size.

```php
/*
 * For the value to be parsed as `auto` it must either be the string `auto` or
 * start with `auto,` after removing the whitespace, followed by other content.
 *
 * @see https://html.spec.whatwg.org/#valid-source-size-list
 * @see https://html.spec.whatwg.org/#parse-a-sizes-attribute
 */
return 1 === preg_match( '/^[ \t\f\r\n]*auto[ \t\f\r\n]*(?:,|$)/i', $sizes_attr );
```

This can also be realized without the PCRE functions, which may have additional perks, including being more legible to some.

```php
function sizes_attribute_is_auto( $value ) {
	$at = strspn( $value, " \t\f\r\n" );
	if ( 0 !== substr_compare( $value, 'auto', $at, 4, true ) ) {
		return false;
	}

	$at += strspn( $value, " \t\f\r\n", $at + 4 );
	return $at === strlen( $value ) || ',' === $value[ $at ];
}
```